spec/requests/net_http_header_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# We can use the redeliver event to test the user-agent header
+RSpec.describe "Net::HTTPHeader sets a default user-agent" do
+  fab!(:admin)
+  fab!(:web_hook)
+  let!(:web_hook_event) { WebHookEvent.create!(web_hook: web_hook, headers: "{}") }
+
+  before do
+    sign_in(admin)
+    stub_request(:post, web_hook.payload_url)
+  end
+
+  it "should set a user-agent when none has been set" do
+    post "/admin/api/web_hooks/#{web_hook.id}/events/#{web_hook_event.id}/redeliver.json"
```

Comment:
Another possible alternative is to create a test only controller endpoint like so: https://github.com/discourse/discourse/blob/c0bbff598a4bf58b29b2e376cdfbc3c2d8921459/spec/requests/net_http_timeout_spec.rb#L8